### PR TITLE
Adding Example project with solidity-coverage lib

### PR DESCRIPTION
**Pull request by stefan-stefanooov**
**Description**:
This PR creates hardhat project with the solidity-coverage plugin in it.

**Related issue(s)**:
https://github.com/hashgraph/hedera-json-rpc-relay/issues/1668
https://github.com/hashgraph/hedera-json-rpc-relay/issues/1491

**Checklist**

- [ ] Documented (Code comments, README, etc.)
- [ ] Tested (unit, integration, etc.)

### CODE REVIEW

File: tools/solidity-coverage-example/.solcover.js [hunk 1]
```diff
@@ -0,0 +1,3 @@
+module.exports = {
```
Comment (Nana-EC) [inline tools/solidity-coverage-example/.solcover.js:1]: Please add license header

File: tools/solidity-coverage-example/README.md [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# Sample Hardhat Project
+
+This project demonstrates a basic Hardhat use case with Hedera. It comes with a sample contract, a test for that contract, and a script that deploys that contract.
```
Comment (Nana-EC) [inline tools/solidity-coverage-example/README.md:3]: The goal here is to show off solidity-coverage not hardhat, so you should update this description

File: tools/solidity-coverage-example/README.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+Try running some of the following tasks:
+
+```shell
+npx hardhat help
+npx hardhat test
+REPORT_GAS=true npx hardhat test
```
Comment (Nana-EC) [inline tools/solidity-coverage-example/README.md:10]: Q: what's this for?

File: tools/solidity-coverage-example/README.md [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+But first the reports have to be generated:
+```shell
+// to run the tests and generate the reports
+npm run test
+// serve the reports on http://127.0.0.1:8080 by default
+npm run coverage:serve
+```
+
+### Limitations:
+There is a limitation around hardhat using our precompiled contracts to a precompiled address.
```
Comment (Nana-EC) [inline tools/solidity-coverage-example/README.md:27]:
```suggestion
There is a limitation around hardhat using our system contracts at their reserved addresses.
```

File: tools/solidity-coverage-example/contracts/TokenSend.sol [hunk 5]
```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: UNLICENSED
```
Comment (Nana-EC) [inline tools/solidity-coverage-example/contracts/TokenSend.sol:1]: Update all the sol files with `Apache-2.0`
```suggestion
// SPDX-License-Identifier: Apache-2.0
```

File: tools/solidity-coverage-example/coverage.json [hunk 6]
```diff
@@ -0,0 +1 @@
...c-relay/tools/solidity-coverage-example/contracts/TokenSend.sol","s":{"1":1,"2":1,"3":1,"4":1,"5":1,"6":1,"7":1},"b":{},"f":{"1":1,"2":1,"3":1,"4":1,"5":1},"fnMap":{"1":{"name":"constructor","line":14,"loc":{"start":{"line":14,"column":4},"end":{"line":16,"column":4}}},"2":{"name":"loadFunds","line":18,"loc":{"start":{"line":18,"column":4},"end":{"line":21,"column":4}}},"3":{"name":"createTransferList","line":23,"loc":{"start":{"line":23,"column":4},"end":{"line":33,"column":4}}},"4":{"name":"storeAmount","line":35,"loc":{"start":{"line":35,"column":4},"end":{"line":37,"column":4}}},"5":{"name":"getAmount","line":39,"loc":{"start":{"line":39,"column":4},"end":{"line":43,"column":4}}}},"statementMap":{"1":{"start":{"line":19,"column":8},"end":{"line":19,"column":102}},"2":{"start":{"line":20,"column":8},"end":{"line":20,"column":40}},"3":{"start":{"line":26,"column":8},"end":{"line":26,"column":131}},"4":{"start":{"line":27,"column":8},"end":{"line":27,"column":107}},"5":{"start":{"line":28,"column":8},"end":{"line":28,"column":113}},"6":{"start":{"line":32,"column":8},"end":{"line":32,"column":42}},"7":{"start":{"line":42,"column":8},"end":{"line":42,"column":27}}},"branchMap":{}}}
```
Comment (Nana-EC) [inline tools/solidity-coverage-example/coverage.json:1]: Q: Why does this file have your filepath?
Does it not pick up relative paths? We should update that

File: tools/solidity-coverage-example/hardhat.config.ts [hunk 7]
```diff
@@ -0,0 +1,10 @@
+/** @type import('hardhat/config').HardhatUserConfig */
```
Comment (Nana-EC) [inline tools/solidity-coverage-example/hardhat.config.ts:1]: Add license header to all `*.ts` and `*.js` files

File: tools/solidity-coverage-example/package.json [hunk 8]
```diff
@@ -0,0 +1,21 @@
+{
+  "name": "solidity-coverage-example",
+  "version": "1.0.0",
+  "description": "Solidity coverage example npm package. It is suppose to figure out what's supported and what's not possible based on the product limitations",
```
Comment (Nana-EC) [inline tools/solidity-coverage-example/package.json:4]:
```suggestion
  "description": "Solidity coverage example npm package. Highlights what's supported and what's not possible based on the product limitations",
```

File: tools/solidity-coverage-example/README.md [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+## Hedera and Hardhat integration
+
+You can find a npm script that launches a http server to serve the code coverage report.
+But first the reports have to be generated:
+```shell
+// to run the tests and generate the reports
+npm run test
+// serve the reports on http://127.0.0.1:8080 by default
+npm run coverage:serve
```
Comment (Nana-EC) [inline tools/solidity-coverage-example/README.md:23]: Can you add a print out or image of what a a successful run looks like maybe for the `Lock.sol`

**Review by Nana-EC — CHANGES_REQUESTED** [review]
Thanks, some suggestions

File: tools/solidity-coverage-example/coverage.json (hunk 6 above)
Comment (stefan-stefanooov) [reply tools/solidity-coverage-example/coverage.json:1]: Now that I look at it again ... Arguably we should not commit this file at all. 
It's the output of the code coverage tool itself. Seems like committing the output of the TypeScript transpilation.

File: tools/solidity-coverage-example/contracts/TokenSend.sol (hunk 5 above)
Comment (stefan-stefanooov) [reply tools/solidity-coverage-example/contracts/TokenSend.sol:1]: Done

File: tools/solidity-coverage-example/hardhat.config.ts (hunk 7 above)
Comment (stefan-stefanooov) [reply tools/solidity-coverage-example/hardhat.config.ts:1]: Done

File: tools/solidity-coverage-example/package.json (hunk 8 above)
Comment (stefan-stefanooov) [reply tools/solidity-coverage-example/package.json:4]: Done

File: tools/solidity-coverage-example/README.md (hunk 9 above)
Comment (stefan-stefanooov) [reply tools/solidity-coverage-example/README.md:23]: Done

File: tools/solidity-coverage-example/README.md (hunk 3 above)
Comment (stefan-stefanooov) [reply tools/solidity-coverage-example/README.md:10]: It's part of the default **README.md** generated by the hardhat initial setup. I'll remove it.

File: tools/solidity-coverage-example/coverage.json (hunk 6 above)
Comment (Nana-EC) [reply tools/solidity-coverage-example/coverage.json]: Great, yeah let's remove it.

File: tools/solidity-coverage-example/cache/solidity-files-cache.json [hunk 10]
```diff
@@ -0,0 +1,279 @@
+{
```
Comment (georgi-l95) [inline tools/solidity-coverage-example/cache/solidity-files-cache.json:1]: remove this file and add dir to gitignore

File: tools/solidity-coverage-example/contracts/PrngSystemContract.sol [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+    address constant PRECOMPILE_ADDRESS = address(0x169);
+
+    function getPseudorandomSeed() external returns (bytes32 seedBytes) {
+        (bool success, bytes memory result) = PRECOMPILE_ADDRESS.call(
+            abi.encodeWithSelector(IPrngSystemContract.getPseudorandomSeed.selector));
+        require(success, "PRNG system call failed");
+        seedBytes = abi.decode(result, (bytes32));
+        emit PseudoRandomSeed(seedBytes);
+    }
+}
```
Comment (georgi-l95) [inline tools/solidity-coverage-example/contracts/PrngSystemContract.sol:19]: add new line

File: tools/solidity-coverage-example/README.md [hunk 12]
```diff
@@ -0,0 +1,29 @@
...
+This project demonstrates a basic usage of the code coverage [plugin](https://github.com/sc-forks/solidity-coverage) of HardHat in a Hedera environment. It comes with a sample contract, a test for that contract, and a script that deploys that contract.
+
+## The integration
+
+You can find a npm script that launches a http server to serve the code coverage report.
+But first the reports have to be generated:
+```shell
+// to run the tests and generate the reports
+npm run test
+// serve the reports on http://127.0.0.1:8080 by default
```
Comment (georgi-l95) [inline tools/solidity-coverage-example/README.md:12]: Why do I need to start a server, just to check the coverage?
Won't a simple `npx hardhat coverage` be enough, just to show the coverage ?

File: tools/solidity-coverage-example/README.md [hunk 13]
```diff
@@ -0,0 +1,29 @@
+# Sample Hedera/Hardhat/Code coverage Project
+
+This project demonstrates a basic usage of the code coverage [plugin](https://github.com/sc-forks/solidity-coverage) of HardHat in a Hedera environment. It comes with a sample contract, a test for that contract, and a script that deploys that contract.
```
Comment (georgi-l95) [inline tools/solidity-coverage-example/README.md:3]:
```suggestion
This project demonstrates a basic usage of the solidity coverage [plugin](https://github.com/sc-forks/solidity-coverage) of HardHat in a Hedera environment. Example comes with simple contract to showcase this functionality, as well as more complex example contract with Hedera Precompile system calls. 
```

File: tools/solidity-coverage-example/contracts/IPrngSystemContract.sol [hunk 14]
```diff
@@ -0,0 +1,8 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity >=0.4.9 <0.9.0;
+
+interface IPrngSystemContract {
+    // Generates a 256-bit pseudorandom seed using the first 256-bits of running hash of n-3 transaction record.
+    // Users can generate a pseudorandom number in a specified range using the seed by (integer value of seed % range)
+    function getPseudorandomSeed() external returns (bytes32);
+}
```
Comment (georgi-l95) [inline tools/solidity-coverage-example/contracts/IPrngSystemContract.sol:8]: new line

**Review by georgi-l95 — CHANGES_REQUESTED** [review]
Overall looks good, we also need to add this to the main tools Readme. Can do it by taking example from The Graph integration table. We need to flag whats possible on Hedera and what is not.

File: tools/solidity-coverage-example/README.md (hunk 12 above)
Comment (stefan-stefanooov) [reply tools/solidity-coverage-example/README.md:12]: The served report has more detailed data for each file.
But that was a good question. I mentioned the stock command in the README as well (with example output)

**Review by Nana-EC — APPROVED** [review]

**Review by AlfredoG87 — APPROVED** [review]
LG

**Review by georgi-l95 — APPROVED** [review]

**Pull request merged**